subcommands/devices/trigger.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/foundriesio/fioctl/subcommands"
+	nanoid "github.com/matoous/go-nanoid/v2"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+)
+
+func init() {
+	triggerCmd := &cobra.Command{
+		Use:   "trigger",
+		Short: "Trigger remote actions on device",
+		Long: `
+*NOTE*: Requires devices running LmP version 97 or later.
+
+This command sets a fioconfig entry that will let a device know it needs to
+run a given trigger *once* and report its results back to the device gateway's
+"tests" API. You can take the "command ID" from this command's output to then
+look up the results using the "fioctl devices tests" command.
+
+# See what triggers are available to the device:
+$ fioctl devices trigger <device>
+
+# Initiate a remote trigger:
+$ fioctl devices trigger <device> <trigger>
+`,
+		Args: cobra.RangeArgs(1, 2),
+		Run:  doTrigger,
+	}
+	triggerCmd.Flags().StringP("reason", "r", "", "The reason for running this command")
+	cmd.AddCommand(triggerCmd)
+}
+
+func loadRemoteActions(d client.DeviceApi) []string {
+	dcl, err := d.ListConfig()
+	subcommands.DieNotNil(err)
+	if len(dcl.Configs) > 0 {
+		for _, cfgFile := range dcl.Configs[0].Files {
+			if cfgFile.Name == "fio-remote-actions" {
+				var actions []string
+				for _, action := range strings.Split(cfgFile.Value, ",") {
+					action = strings.TrimSpace(action)
+					if len(action) > 0 {
+						actions = append(actions, action)
+					}
+				}
+				if actions == nil {
+					break
+				}
+				return actions
+			}
+		}
+	}
+	fmt.Println("Remote actions are not configured for this device")
+	os.Exit(0)
+	return nil
+}
+
+func doTrigger(cmd *cobra.Command, args []string) {
+	name := args[0]
+
+	// Quick sanity check for device
+	d := getDevice(cmd, name)
+
+	// See what triggers are allowed
+	allowed := loadRemoteActions(d.Api)
+
+	if len(args) == 1 {
+		fmt.Println("Available actions:")
+		for _, trigger := range allowed {
+			fmt.Println("*", trigger)
+		}
+		return
+	}
+
+	action := args[1]
+	if !slices.Contains(allowed, action) {
+		subcommands.DieNotNil(fmt.Errorf("Invalid action: %s", action))
+	}
+	reason, _ := cmd.Flags().GetString("reason")
+
+	// For the random part anything >= 15 characters will give us about 1,000
+	// years before a collision (assuming 100 IDs per second):
+	//    https://zelark.github.io/nano-id-cc/
+	idLen := 15
+	maxActionLen := 48 - idLen - 1  // 1 for the underscore in the CommandId below
+	if len(action) > maxActionLen { // Device Gateway has max len of 48
+		subcommands.DieNotNil(fmt.Errorf("Action name(%s) too long. Max length is %d", action, maxActionLen))
+	}
+	id, err := nanoid.New(idLen)
+	subcommands.DieNotNil(err)
+
+	opts := triggerOptions{
+		Capture:   true,
+		Command:   action,
+		CommandId: fmt.Sprintf("%s_%s", action, id),
+		Reason:    reason,
+	}
+
+	ccr := opts.AsConfig()
+	subcommands.DieNotNil(d.Api.PatchConfig(ccr, false))
+	fmt.Println("Config change submitted. Command ID is:", opts.CommandId)
```

Comment:
Can we print here how the user can see the command log/result? Is there a fioctl command for that?